File: OffByOne/src/OffByOne.LanguageCore/Ast/Literals/MoneyLiteral.cs
namespace OffByOne.LanguageCore.Ast.Literals
{
    using OffByOne.LanguageCore.Ast.Literals.Base;
    using OffByOne.LanguageCore.Visitors.Contracts;

    public class MoneyLiteral : Literal
    {
        public MoneyLiteral(decimal value)
        {
            this.Value = value;
        }

        public MoneyLiteral(string value)
            : this(decimal.Parse(value))
        {
        }

        public decimal Value { get; private set; }

        public override TResult Accept<TResult, TContext>(
            ILiteralVisitor<TResult, TContext> visitor,
            TContext context)
        {
            return visitor.Visit(this, context);
        }
    }
}

File: vanVlaanderenDerksen/QLJava/src/sc/ql/model/expressions/literals/MoneyLiteral.java
package sc.ql.model.expressions.literals;

import java.math.BigDecimal;

import sc.ql.model.Expression;
import sc.ql.model.visitors.ExpressionVisitor;

public class MoneyLiteral extends Expression {
	private final BigDecimal value;
	
	public MoneyLiteral(BigDecimal value) {
		this.value = value;
	}
	
	public BigDecimal getValue() {
		return this.value;
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
}
